static/packages/autocomplete.js

```diff
@@ -17,124 +17,97 @@ class AutocompleteField {
+            city = components[0];
         else {
             if (components[1].charAt(0) === ' ')
                 components[1] = components[1].substring(1);
 
-            address['city'] = components[1];
-            streetDetails = self.extractStreetDetails(components[0].split(' '));
-
-            address['street'] = streetDetails[0];
-            address['street_number'] = streetDetails[1];
-        }
-
-        if (typeof address['street'] === 'undefined')
-            address['street'] = '<no street>';
-        if (typeof address['street_number'] === 'undefined' || isNaN(address['street_number']))
-            address['street_number'] = 1;
-
-        address['formatted'] = customInput;
-
-        return address;
-    }
-
-    extractStreetDetails(self, detailsStr) {
-        let word = "";
-        let streetName = "";
-        let streetNumber;
-
-        streetDetails = detailsStr;
+            if (streetNumber = + components[0].match(numRegex)) {
+                street = components[0];
+                city = components[1];
+            }
+            else {
+                streetNumber = +components[1].match(numRegex);
+                street = components[1];
+                city = components[0];
+            }
 
-        for (i = 0; !self.isNumber(word); i++) {
-            word = streetDetails[i];
-            streetName += streetDetails + ' ';
+            street = street.replace(/[0-9]/g, '').slice(0, -1);
+            alert("'" + String(street) + "'");
         }
 
-        streetName = streetName.slice(0, -1);
-        streetNumber = parseInt(word);
+        street ??= '<no street>';
 
-        return [streetName, streetNumber];
-    }
-
-    isNumber(str) {
-        num = parseInt(str);
+        if (typeof streetNumber === 'undefined' || streetNumber == '')
+            streetNumber = 1;
 
-        return !isNaN(num);
+        self.formatted = customInput;
+        self.city = city;
+        self.street = street;
+        self.street_number = streetNumber;
     }
 
-    extractAddressComponents(place) {
-        let address = {};
-        let str = "";
+    extractAddressComponents(self, place) {
 
         for (const component of place.address_components) {
             const componentType = component.types[0];
 
             switch (componentType) {
-                case "street_number": {
-                    let streetNumber = component.long_name;
-                    address['street_number'] = streetNumber;
-                    str = streetNumber;
+                case "street_number":
+                    self.streetNumber = component.long_name;
                     break;
-                }
 
-                case "route": {
-                    let streetName = component.short_name;
-                    address['street'] = streetName;
-                    str = streetName + ' ' + str + ', ';
+                case "route":
+                    self.streetName = component.short_name;

```

Comment:
You use `address.street` above.

Reply:
Yup a bit of confusion cuz js allows creating props this way smh